src/com/projectkorra/projectkorra/airbending/flight/FlightMultiAbility.java

```diff
@@ -97,6 +101,15 @@ public FlightMultiAbility(final Player player) {
 			return;
 		}
 
+		this.allowedFlightModes.put(FlightMode.SOAR, player.hasPermission("bending.ability.flight.soar"));
+		this.allowedFlightModes.put(FlightMode.GLIDE, player.hasPermission("bending.ability.flight.glide"));
+		this.allowedFlightModes.put(FlightMode.LEVITATE, player.hasPermission("bending.ability.flight.levitate"));
+		this.firstAvaliableFlightMode = this.allowedFlightModes
+			.keySet()
+			.stream()
+			.filter(mode -> this.allowedFlightModes.get(mode)).findFirst();
```

Comment:
While I like what you are trying to achieve, I think this sort of thing should be implemented for multiabilities as a whole. Right now, WaterArms supports permissions for each subability, but they aren't removed from the list. So I think this functionality would be better suited to be fused into Multiabilities as a whole. And then WaterArms and flight are just updated to use the new system.